### ISSUE TITLE
Improving the Describe > Specific > Boxplot dialogue - jitter plot defaults

### ISSUE THREAD

**Issue by rdstern** [IDEMSInternational/R-Instat#4537]
There are a number of issues with this dialogue. This issue is concerned with the defaults for the jitter plot component.  
1. In the instructions on jitter plots it says that if you don't give a value for the horizontal jitter, then it defaults to 40% (i.e. 0.4).  I would prefer (our) default to be smaller than this and suggest 0.2.
2. The vertical jitter also defaults to 0.4 and I would prefer our default be 0, i.e. no vertical jitter.

These will have to be given in the command when jitter is used. 

In the section where the changes can be made for this layer, i.e. the sub-dialogue choosing jitter, then our defaults should be given there, i.e. 0.2 and 0.

In addition the limit for the jitter is 0.5.  So changing in steps of 0.1 is quite large.  I suggest we change in steps of 0.05 with a bottom limit of 0 and a top limit of 0.5.
Could there also be either a tool-tip or a label  e.g. (0 gives no jitter, maximum is 0.5).

Once this is done could you look also at the **Describe > One Variable > Graphics** so the jitter plot here and on the boxplot dialogue are consistent.

**Comment by maxwellfundi** [comment]
@rdstern  This could actually be fixed easily by adding these parameters to the violin and jitter functions when they are selected and updating the our layer parameters too.

**Comment by dannyparsons** [comment]
Yes we could add options for these on the main dialog when jitter is selected. They would be synced with the sub dialogs automatically if done correctly.

For the values, it does default to 40% for horizontal/vertical but that is not equal to setting them to 0.4. The amount of jitter is specified in units on the axis - so it depends on the data. If the parameter is not given, ggplot does a calculation to give a jitter equivalent to 40%, but 0.4 would be 0.4 units on the axis. So shouldn't have 0.4 as the default. Since they can be any values in general I suggest textboxes for these. There should be checkboxes for these to override the default so that if unchecked you do get the 40% amount.

**Comment by rdstern** [comment]
I am not quite clear on this.  Currently the jitter plot is part of the boxplot dialogue, where the x-axis is a factor, and there (I think) the 0.4  and 40% are the same thing?  I realise that (in general) we can use jitter plots for other than x being a factor, but we can't do that in this dialogue.

**Comment by dannyparsons** [comment]
Ah yes that's right for factors the levels are taken as integer so 0.4 is 40%. 
So 40% can be the default for horizontal. For the vertical jitter this depends on the y axis scale.

**Comment by rdstern** [comment]
That's where I don't like 40% for horizontal.  That's the same as the size of a boxplot or violin plot.  So you don't see them.
And I suggest it is confusing - and not always necessary to have any vertical jitter.  And often it will be confusing.  So my default for vertical is zero.  That's also consistent with CAST.

**Comment by maxwellfundi** [comment]
@dannyparsons  what am adding here now is the new parameters to the jitter. Do we need nuds where the users can change or we can just have these parameters added automatically when jitter is selected.

**Comment by rdstern** [comment]
Either is possible.  I was assuming the second, i.e. "just have these parameters added automatically when jitter is selected".  These will be sensible for the case here (i.e. the x-s are a factor and keep to our principle of having the main dialogue as simple as possible.
At the same time, could you explore one I missed,  in my list of items for these dialogues.  That is to make the outlier colour red for boxplots also as the default.  Then when we have both we know where they are from.

**Comment by maxwellfundi** [comment]
Adding outlier colour as red could also be added automatically and is easy too. I wonder if you would like it to apply to all the three plots or just the boxplot

**Comment by rdstern** [comment]
It is just the boxplot.  That's the only one that has **outlier colour** as an argument

**Comment by dannyparsons** [comment]
The vertical jitter can't be a nud because it could be any scale depending on the data. The horizontal jitter is usually going to be between 0 and 1 but I think they could be both be textboxes to make it easier.

I now think we should have something like "Auto" and "Fixed" radio buttons side by side - one set for horizontal one for vertical. Where "Auto" removes the parameter and does the ggplot default and "Fixed" is a fixed value. That gives all the possibilities of ggplot. 
We can have whatever we like for our own default which I think Roger is suggesting "Fixed 0.2" for horizontal and "Fixed 0" for vertical.

**Comment by lilyclements** [comment]
Closing as we have changed this dialog considerably since 2018, and improved beyond these discussions
